public/app/diarization-timeline.test.js

```
@@ -0,0 +1,181 @@
+
+  // ─ 2 ─
+  it('accepts diarization result with correct segment count', () => {
+    expect(() => onDiarizationResult({ segments: SAMPLE_SEGMENTS, duration: 20 })).not.toThrow();
+  });
+
+  // ─ 3 ─
+  it('builds speaker registry from segments', () => {
+    onDiarizationResult({ segments: SAMPLE_SEGMENTS, duration: 20 });
+    // Access internal state via re-init trick — check DOM side effects
+    // After processing 2 unique speakers S1, S2 should exist
+    const cards = document.querySelectorAll('.speaker-card');
```

ESLint

[error] 'cards' is assigned a value but never used.

(no-unused-vars)